x-pack/plugins/lists/server/services/exception_lists/exception_list_client.ts

```diff
@@ -865,4 +904,257 @@ export class ExceptionListClient {
       user,
     });
   };
+
+  /**
+   * Opens a point in time (PIT) for either exception lists or exception list items.
+   * See: https://www.elastic.co/guide/en/elasticsearch/reference/current/point-in-time-api.html
+   * @params namespaceType {string} "agnostic" or "single" depending on which namespace you are targeting
+   * @params options {Object} The saved object PIT options
+   * @return {SavedObjectsOpenPointInTimeResponse} The point in time (PIT)
+   */
+  public openPointInTime = async ({
+    namespaceType,
+    options,
+  }: OpenPointInTimeOptions): Promise<SavedObjectsOpenPointInTimeResponse> => {
+    const { savedObjectsClient } = this;
+    return openPointInTime({
+      namespaceType,
+      options,
+      savedObjectsClient,
+    });
+  };
+
+  /**
+   * Closes a point in time (PIT) for either exception lists or exception list items.
+   * See: https://www.elastic.co/guide/en/elasticsearch/reference/current/point-in-time-api.html
+   * @params pit {string} The point in time to close
+   * @return {SavedObjectsOpenPointInTimeResponse} The point in time (PIT)
+   */
+  public closePointInTime = async ({
+    pit,
+  }: ClosePointInTimeOptions): Promise<SavedObjectsClosePointInTimeResponse> => {
+    const { savedObjectsClient } = this;
+    return closePointInTime({
+      pit,
+      savedObjectsClient,
+    });
+  };
+
+  /**
+   * Finds an exception list item within a point in time (PIT) and then calls the function
+   * `executeFunctionOnStream` until the maxPerPage is reached and stops.
+   * NOTE: This is slightly different from the saved objects version in that it takes
+   * an injected function, so that we avoid doing additional plumbing with generators
+   * to try to keep the maintenance of this machinery simpler for now.
+   *
+   * If you want to stream all results up to 10k into memory for correlation this would be:
+   * @example
+   * ```ts
+   * const exceptionList: ExceptionListItemSchema[] = [];
+   * const executeFunctionOnStream = (response: FoundExceptionListItemSchema) => {
+   *   exceptionList = [...exceptionList, ...response.data];
+   * }
+   * await client.findExceptionListItemPointInTimeFinder({
+   *   filter,
+   *   executeFunctionOnStream,
+   *   namespaceType,
+   *   maxSize: undefined, // NOTE: This is unbounded when it is "undefined"
```

Comment:
For this example, this should be 10,000 I think.

Reply:
Right now we aren't bounding any of the `maxSize` in the code base. But do you think we should or at least suggest in the documentation to bound their `maxSize`?